### PR TITLE
If the request store is empty, generate an empty Object response

### LINKED ISSUE #330: Exception when 'preferences' are `null`
My client is throwing:

```
TypeError: Cannot convert undefined or null to object
```

On this line:
https://github.com/mozilla-magnet/magnet-client/blob/b77a61e/lib/app.js#L196

+@samgiles

### PR DESCRIPTION
**Pull request by samgiles**
Closes #330 

@wilsonpage r?

### CODE REVIEW

**Comment by wilsonpage** [comment]
This is still throwing. I think empty JSON objects resolve to `null` somewhere in the pipeline. Either we cope with `null` on the client, or find a way to send an empty object.

**Comment by samgiles** [comment]
@wilsonpage I couldn't reproduce the null issue.  I've added a fallback in the JavaScript.

**Comment by wilsonpage** [comment]
@samgiles I can reproduce on emulator and on device. Have you tried uninstalling the app and re-installing fresh?

File: android/app/src/main/java/org/mozilla/magnet/magnetapi/ApiPreferences.java [hunk 1]
```diff
@@ -28,7 +29,13 @@
...
     @Override
     public void get(String aPath, Callback aCallback) {
-        aCallback.resolve(getCache().getJsonObject(PREFERENCES_STORE_KEY));
+        JSONObject json = getCache().getJsonObject(PREFERENCES_STORE_KEY);
+
+        if (json == null) {
+          json = new JSONObject();
+        }
+
+        aCallback.resolve(json);
```
Comment (wilsonpage) [inline android/app/src/main/java/org/mozilla/magnet/magnetapi/ApiPreferences.java:37]: Should we remove this code if it's not fixing the issue?

**Comment by samgiles** [comment]
r? I got to the bottom of it.

Since Wilson's jet set, @arcturus?

**Comment by arcturus** [comment]
r+ once travis is green

**Comment by wilsonpage** [comment]
@samgiles can you rebase this?

**Pull request merged**